extensions/ql-vscode/src/helpers.ts

```diff
@@ -29,8 +29,34 @@ export async function showAndLogErrorMessage(message: string, {
   items = [] as string[],
   fullMessage = undefined as (string | undefined)
 } = {}): Promise<string | undefined> {
-  return internalShowAndLog(message, items, outputLogger, Window.showErrorMessage, fullMessage);
+  return internalShowAndLog(shortenErrorMessage(message), items, outputLogger, Window.showErrorMessage, fullMessage);
 }
+
+/**
+ * Creates a succint version of an error message, e.g. for displaying in a pop-up.
+ * @param errorMessage The error message to shorten.
+ * @return A shortened version of the error message.
+ */
+function shortenErrorMessage(errorMessage: string): string {
+  // Filter out lines corresponding to Java stack traces.
+  const stackTraceLine = /\n\s*(com|org|net)\..*\.java:[0-9]+\)(?=\n)/g;
+  let succintMessage = errorMessage.toString().replace(stackTraceLine, '');
+
+  // Filter out lines corresponding to log lines.
+  const logLine = /\n\[[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9] [0-9][0-9]:[0-9][0-9]:[0-9][0-9]\].*/g;
+  succintMessage = succintMessage.replace(logLine, '');
+
+  // Trim indentation at the start of lines.
+  const indentedLine = /\n\s+/g;
+  succintMessage = succintMessage.replace(indentedLine, '\n');
+
+  // Remove duplicated lines.
+  const duplicatedLine = /\n(.*)\n\1/g;
```

Comment:
Careful here. The `\n` won't work on windows. You should probably be working on this line-by-line instead of running the regex over each line.